### PR TITLE
Creates a distinction between read and write variable and field access

### PR DESCRIPTION
**Pull request by GerardPaligot**
No issue.

### CODE REVIEW

File: src/test/java/spoon/test/variable/AccessTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+								  new AbstractFilter<CtArrayAccess<?, CtExpression<?>>>(
+										  CtArrayAccess.class) {
+									  @Override
+									  public boolean matches(
+											  CtArrayAccess<?, CtExpression<?>> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(arraysRead.size() + arraysWrite.size(), arraysAccess.size());
```
Comment (monperrus) [inline src/test/java/spoon/test/variable/AccessTest.java]: better to have a literal, assertEquals(2, arrayAccess.size())

File: src/test/java/spoon/test/variable/AccessTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		final List<CtFieldAccess<?>> fieldsAccess =
+				Query.getElements(factory,
+								  new AbstractFilter<CtFieldAccess<?>>(CtFieldAccess.class) {
+									  @Override
+									  public boolean matches(CtFieldAccess<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(fieldsRead.size() + fieldsWrite.size(), fieldsAccess.size());
```
Comment (monperrus) [inline src/test/java/spoon/test/variable/AccessTest.java]: idem

File: src/test/java/spoon/test/variable/AccessTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		final List<CtVariableAccess<?>> variablesAccess =
+				Query.getElements(factory,
+								  new AbstractFilter<CtVariableAccess<?>>(CtVariableAccess.class) {
+									  @Override
+									  public boolean matches(CtVariableAccess<?> element) {
+										  return super.matches(element);
+									  }
+								  });
+
+		assertEquals(variablesRead.size() + variablesWrite.size(), variablesAccess.size());
```
Comment (monperrus) [inline src/test/java/spoon/test/variable/AccessTest.java]: idem

File: src/test/java/spoon/test/variable/AccessTest.java (hunk 1 above)
Comment (GerardPaligot) [inline src/test/java/spoon/test/variable/AccessTest.java]: I disagree. The size of `CtArrayAccess` (the same for `CtVariableAccess` and `CtFieldAccess`) must be the sum of `CtArrayRead` and `CtArrayWrite`.

**Pull request merged**